File: backend/src/main/java/com/example/outletmanagement/service/UserService.java
package com.example.outletmanagement.service;

import com.example.outletmanagement.payload.dto.request.RegisterRequest;
import com.example.outletmanagement.payload.dto.request.UserCreateRequest;
import com.example.outletmanagement.payload.dto.response.UserResponse;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

public interface UserService {
    UserResponse createUser(UserCreateRequest request);
    Page<UserResponse> getAllUsers(Pageable pageable);
    UserResponse getUserById(Long id);
    UserResponse updateUserRole(Long id, String role);
    void deleteUser(Long id);
}

File: outlet-frontend/src/services/userService.js
import apiClient from '../api/apiClient';

export const userService = {
  getAllUsers: async (page = 0, size = 10) => {
    const response = await apiClient.get(`/api/users?page=${page}&size=${size}`);
    return response.data;
  },

  createUser: async (userData) => {
    const response = await apiClient.post('/api/users', userData);
    return response.data;
  },

  updateUserRole: async (id, role) => {
    const response = await apiClient.patch(`/api/users/${id}/role?role=${role}`);
    return response.data;
  },

  deleteUser: async (id) => {
    await apiClient.delete(`/api/users/${id}`);
  }
};
